controller/src/main/java/com/epam/esm/contoller/OrderController.java

```diff
@@ -0,0 +1,99 @@
+package com.epam.esm.contoller;
+
+import com.epam.esm.dto.ActionHypermedia;
+import com.epam.esm.dto.CreateActionHypermedia;
+import com.epam.esm.dto.OrderDTO;
+import com.epam.esm.service.OrderService;
+import com.epam.esm.util.builder.ActionHypermediaLinkBuilder;
+import com.epam.esm.util.builder.CreateHypermediaLinkBuilder;
+import com.epam.esm.util.builder.OrderLinkBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+
+    private static final int DEFAULT_LIMIT_INT = 5;
+    private static final int DEFAULT_PAGE_INT = 1;
+    private static final String DEFAULT_LIMIT = "5";
+    private static final String DEFAULT_PAGE = "1";
+    private static final String ERROR_MESSAGE = "error";
+
+    @Autowired
+    private OrderService service;
+
+    @GetMapping("/")
+    public ResponseEntity<?> retrieveAllOrders(@RequestParam(defaultValue = DEFAULT_LIMIT) int limit,
+                                               @RequestParam(defaultValue = DEFAULT_PAGE) int page) {
+        try {
+            List<OrderDTO> resultList = service.findAll(limit, page);
+            for (int i = 0; i < resultList.size(); i++) {
```

Comment:
I suppose we could use java 8 here, something like this:
~~~
        List<OrderDTO> resultList = service.findAll(limit, page)
                .stream()
                .map(it -> new OrderLinkBuilder(it)
                        .buildCertificateReferenceLink().buildUserReferenceLink().buildRetrieveSpecificOrderLink()
                        .getHypermedia()
                ).collect(Collectors.toList());
~~~